homeassistant/components/min_max/sensor.py

```diff
@@ -290,61 +288,80 @@ def _async_min_max_sensor_state_listener(
+        entity: str = event.data["entity_id"]
 
-        if self._unit_of_measurement is None:
-            self._unit_of_measurement = new_state.attributes.get(
+        states: list[tuple[str, State]] = [
+            (entity_id, state)
+            for entity_id in self._entity_ids
+            if (state := self.hass.states.get(entity_id))
+        ]
+        for entity_id, new_state in states:
+            if (
+                new_state is None
+                or new_state.state is None
+                or new_state.state
+                in [
+                    STATE_UNKNOWN,
+                    STATE_UNAVAILABLE,
+                ]
+            ):
+                calc_states[entity_id] = STATE_UNKNOWN
+                if not update_state:
+                    self._calc_values(calc_states)
+                    return
+            try:
+                calc_states[entity_id] = float(new_state.state)
+                if entity_id in self._state_incorrect:
+                    self._state_incorrect.remove(entity_id)
+            except ValueError:
+                calc_states[entity_id] = STATE_UNKNOWN
+                if entity_id not in self._state_incorrect:
+                    self._state_incorrect.append(entity_id)
+                    _LOGGER.warning(
+                        "Unable to store state for entity %s with state %s. Only numerical states are supported",
+                        entity_id,
+                        new_state.state,
+                    )
+
+        if entity_state:
+            try:
+                self.last = float(entity_state.state)
+                self.last_entity_id = entity
+            except ValueError:
+                pass
+
+            if self._unit_of_measurement is None:
+                self._unit_of_measurement = entity_state.attributes.get(
+                    ATTR_UNIT_OF_MEASUREMENT
+                )
+
+            if self._unit_of_measurement != entity_state.attributes.get(
                 ATTR_UNIT_OF_MEASUREMENT
-            )
-
-        if self._unit_of_measurement != new_state.attributes.get(
-            ATTR_UNIT_OF_MEASUREMENT
-        ):
-            _LOGGER.warning(
-                "Units of measurement do not match for entity %s", self.entity_id
-            )
-            self._unit_of_measurement_mismatch = True
-
-        try:
-            self.states[entity] = float(new_state.state)
-            self.last = float(new_state.state)
-            self.last_entity_id = entity
-        except ValueError:
-            _LOGGER.warning(
-                "Unable to store state. Only numerical states are supported"
-            )
+            ):
+                if self._unit_of_measurement_mismatch is False:
+                    _LOGGER.warning(
+                        "Units of measurement do not match for entity %s",
+                        self.entity_id,
+                    )
+                self._unit_of_measurement_mismatch = True
```

Comment:
Move this to a new method `update_attributes` which returns without doing anything if `entity_state` is `None` to avoid the nesting.